skbio/tree/tests/test_tree.py

```diff
@@ -1337,6 +1337,101 @@ def test_support(self):
         for node in tree.children:
             self.assertIsNone(node.support())
 
+    def test_unpack(self):
+        """Unpack an internal node."""
+        # test unpacking a node without branch length
+        tree = TreeNode.read(['((c,d)a,(e,f)b);'])
+        tree.find('b').unpack()
+        exp = '((c,d)a,e,f);\n'
+        self.assertEqual(str(tree), exp)
+
+        # test unpacking a node with branch length
+        tree = TreeNode.read(['((c:2.0,d:3.0)a:1.0,(e:2.0,f:1.0)b:2.0);'])
+        tree.find('b').unpack()
+        exp = '((c:2.0,d:3.0)a:1.0,e:4.0,f:3.0);'
+        self.assertEqual(str(tree).rstrip(), exp)
+
+        # test attempting to unpack root
+        tree = TreeNode.read(['((d,e)b,(f,g)c)a;'])
+        msg = 'Cannot unpack root.'
+        with self.assertRaisesRegex(ValueError, msg):
+            tree.find('a').unpack()
+
+        # test attempting to unpack tip
+        msg = 'Cannot unpack tip.'
+        with self.assertRaisesRegex(ValueError, msg):
+            tree.find('d').unpack()
```

Comment:
I am ok to keep it; but it's usually a good practice to atomize the tests into separate functions.